Sources/SwiftParser/TokenSpecSet.swift

```diff
@@ -21,6 +21,38 @@ protocol TokenSpecSet: CaseIterable {
   init?(lexeme: Lexer.Lexeme)
 }
 
+// A way to combine two token spec sets into an aggregate token spec set.
+enum EitherTokenSpecSet<LHS: TokenSpecSet, RHS: TokenSpecSet>: TokenSpecSet {
+  case lhs(LHS)
+  case rhs(RHS)
+
+  init?(lexeme: Lexer.Lexeme) {
+    if let x = LHS.init(lexeme: lexeme) {
+      self = .lhs(x)
+      return
+    }
+    if let y = RHS.init(lexeme: lexeme) {
+      self = .rhs(y)
+      return
+    }
```

Comment:
Nit: You don’t need `.init`

```suggestion
    if let x = LHS(lexeme: lexeme) {
      self = .lhs(x)
      return
    }
    if let y = RHS(lexeme: lexeme) {
      self = .rhs(y)
      return
    }
```